File: 693/java/693.java
class Solution {
    public boolean hasAlternatingBits(int n) {
        int x = n^(n>>1);
        return (x&(x+1))==0;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        System.out.println(sol.hasAlternatingBits(5)); // true
        System.out.println(sol.hasAlternatingBits(7)); // false
        System.out.println(sol.hasAlternatingBits(11)); // false
        System.out.println(sol.hasAlternatingBits(10)); // true
    }
}

File: 693/python/693.py
class Solution:
    def hasAlternatingBits(self, n: int) -> bool:
        x = n^(n>>1)
        return x&(x+1)==0
    
if __name__ == "__main__":
    s = Solution()
    print(s.hasAlternatingBits(5)) # true
    print(s.hasAlternatingBits(7)) # false
    print(s.hasAlternatingBits(11)) # false
    print(s.hasAlternatingBits(10)) # true